File: 백준/Silver/15649. N과 M （1）/N과 M （1）.js
const [N, M] = require('fs').readFileSync(0, 'utf-8').toString().trim().split(' ').map(Number);
const dx = Array.from(Array(N), (_, i) => i + 1);
let answer = '';

function dfs(x, sequence, visited) {
    if (sequence.length === M) {
        answer += `${sequence.join(' ')}\n`;
        return;
    }
    
    for (let nx of dx) {
        if (visited.has(nx)) continue;
        
        sequence.push(nx);
        visited.add(nx);
        dfs(nx, sequence, visited);
        visited.delete(nx);
        sequence.pop();
    }
}

for (let x of dx) {
    const sequence = [x];
    const visited = new Set();
    
    visited.add(x);
    
    dfs(x, sequence, visited);
}

console.log(answer);

File: 백준/Silver/15649. N과 M （1）/N과 M （1）.py
N, M = map(int, input().split())
visited = set()

def dfs(x, seq):
    visited.add(x)
    seq.append(x)
    
    if len(seq) == M:
        print(*seq)
    
    for v in range(1, N+1):
        if v in visited:
            continue

        dfs(v, seq)
        
    seq.pop()
    visited.remove(x)
        
for i in range(1, N+1):
    dfs(i, [])
   